src/noisepy/seis/scedc_s3store.py

```diff
@@ -27,20 +27,22 @@ class SCEDCS3DataStore(RawDataStore):
     # for checking the filename has the form: CIGMR__LHN___2022002.ms
     file_re = re.compile(r".*[0-9]{7}\.ms$", re.IGNORECASE)
 
-    def __init__(self, directory: str, chan_catalog: ChannelCatalog, channel_filter: Callable[[Channel], bool] = None):
+    def __init__(self, path: str, chan_catalog: ChannelCatalog, channel_filter: Callable[[Channel], bool] = None):
         """
         Parameters:
-            directory: directory to look for ms files
+            path: path to look for ms files. Can be a local file directory or an s3://... path
             chan_catalog: ChannelCatalog to retrieve inventory information for the channels
             channel_filter: Function to decide whether a channel should be used or not,
                             if None, all channels are used
         """
         super().__init__()
-        self.directory = directory
+        self.fs = get_filesystem(path)
```

Comment:
I was just thinking, is it worth to check for the correct scheme of `s3` if the path is a URL? Seem like that's important if this module is meant to work with `s3` ONLY and not other file systems?

Reply:
No, it's not meant to work exclusively with S3. It only happens to be where we have data now, so I don't think we need to constrain it.

Reply:
Okay. In that case, maybe need to rename the module to be `scedc_store` and remove references to just S3?

Reply:
Oh, I see what you mean now, yes, the scedc_s3store is specific to S3 and the file naming conventions used there by SCEDC. However, I can see somebody mirroring this data (or a portion) with a different provider (eg. Azure blobs), just like you can download some of it locally. In this case I see more value in allowing different fsspec providers to be used. The class is still meant to follow the schema/naming of the original S3 bucket though, so I think it makes sense for the name to reflect that.

Reply:
Gotcha. Okay. In that case this module is appropriately named. I think an explanation somewhere about this naming scheme would be helpful for a user. Thanks for clearing that up 😄